clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -32,16 +33,39 @@ void processRecordDecl(RecordDecl *Declaration, ProgramInfo &Info,
       SourceManager &SM = Context->getSourceManager();
       FileID FID = FL.getFileID();
       const FileEntry *FE = SM.getFileEntryForID(FID);
+
+      //detect whether this RecordDecl is part of an inline struct
+      bool IsInLineStruct = false;
+      Decl *D = Declaration->getNextDeclInContext();
+      if (VarDecl *VD = dyn_cast_or_null<VarDecl>(D)) {
+        auto VarTy = VD->getType();
+        unsigned int BeginLoc = VD->getBeginLoc().getRawEncoding();
+        unsigned int EndLoc = VD->getEndLoc().getRawEncoding();
+        IsInLineStruct = !(VarTy->isPointerType() || VarTy->isArrayType()) &&
+                         !VD->hasInit() &&
+                         lastRecordLocation >= BeginLoc &&
+                         lastRecordLocation <= EndLoc;
+      }
       if (FE && FE->isValid()) {
         // We only want to re-write a record if it contains
         // any pointer types, to include array types.
-        for (const auto &D : Definition->fields()) {
-          if (D->getType()->isPointerType() || D->getType()->isArrayType()) {
-            if(FL.isInSystemHeader() || Definition->isUnion()) {
-              CVarSet C = Info.getVariable(D, Context);
-              std::string Rsn = "External struct field or union encountered";
-              CB.constraintAllCVarsToWild(C, Rsn, nullptr);
-            }
+        for (const auto &F : Definition->fields()) {
+          auto FieldTy = F->getType();
+          // If the RecordDecl is a union or in a system header
+          // and this field is a pointer, we need to mark it wild;
+          bool FieldInUnionOrSysHeader =
+              (FL.isInSystemHeader() || Definition->isUnion()) &&
+              (FieldTy->isPointerType() || FieldTy->isArrayType());
```

Comment:
You don't need the check about pointer types. You are checking that below. You just need the isUnion or inSysHeader parts.

Reply:
Yeah, I noticed the similarity between the two booleans and refactored it into the if statement. Looks like my refactoring didn't go all the way through 🤦